skimage/filters/thresholding.py

```diff
@@ -719,3 +722,165 @@ def threshold_triangle(image, nbins=256):
+    times w centered around the pixel. k is a configurable parameter
+    that weights the effect of standard deviation.
+
+    Parameters
+    ----------
+    image: (N, M) ndarray
+        Grayscale input image.
+    window_size : int, optional
+        Odd size of pixel neighborhood window (e.g. 3, 5, 7...).
+    k : float, optional
+        Value of parameter k in threshold formula.
+
+    Returns
+    -------
+    threshold : (N, M) ndarray
+        Threshold mask. All pixels with an intensity higher than
+        this value are assumed to be foreground.
+
+    Notes
+    -----
+    This algorithm is originally designed for text recognition.
+
+    References
+    ----------
+    .. [1] Niblack, W (1986), An introduction to Digital Image
+           Processing, Prentice-Hall.
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> image = data.page()
+    >>> binary_image = threshold_niblack(image, window_size=7, k=0.1)
+    """
+    assert_nD(image, 2)
+    m, s = _mean_std(image, window_size)
+    return m - k * s
+
+
+def threshold_sauvola(image, window_size=15, k=0.2, r=None):
+    """Applies Sauvola local threshold to an array. Sauvola is a
+    modification of Niblack technique.
+
+    In the original method a threshold T is calculated for every pixel
+    in the image using the following formula:
+
+    T = m(x,y) * (1 + k * ((s(x,y) / R) - 1))
+
+    where m(x,y) and s(x,y) are the mean and standard deviation of
+    pixel (x,y) neighborhood defined by a rectangular window with size w
+    times w centered around the pixel. k is a configurable parameter
+    that weights the effect of standard deviation.
+    R is the maximum standard deviation of a greyscale image.
+
+    Parameters
+    ----------
+    image: (N, M) ndarray
+        Grayscale input image.
+    window_size : int, optional
+        Odd size of pixel neighborhood window (e.g. 3, 5, 7...).
+    k : float, optional
+        Value of the positive parameter k.
+    r : float, optional
+        Value of R, the dynamic range of standard deviation.
+        If None, set to the half of the image dtype range.
+    offset : float, optional
+        Constant subtracted from obtained local thresholds.
+
+    Returns
+    -------
+    threshold : (N, M) ndarray
+        Threshold mask. All pixels with an intensity higher than
+        this value are assumed to be foreground.
+
+    Notes
+    -----
+    This algorithm is originally designed for text recognition.
+
+    References
+    ----------
+    .. [1] J. Sauvola and M. Pietikainen, "Adaptive document image
+           binarization," Pattern Recognition 33(2),
+           pp. 225-236, 2000.
+           DOI:10.1016/S0031-3203(99)00055-2
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> image = data.page()
+    >>> binary_sauvola = threshold_sauvola(image,
+    ...                                    window_size=15, k=0.2)
+    """
+    assert_nD(image, 2)
```

Comment:
As above, please add n-dimensional support to `_mean_std` and remove this check. I'm happy to help if you have questions about how to do it!